### PR TITLE
[PO-2994] Release Exception/Unapproved Deployments report

### PR DESCRIPTION
**Pull request by popovm**
Creates a report page of unapproved deployments and release exception for a given app.

Original ticket: https://jira.fundingcircle.com/browse/GQ-30

The new page is accessible from Releases -> choose app from the list -> click "Unapproved Releases and Release Exception" for chosen app

There is a datepicker for the time period. 
The first table shows all unapproved deployments.
The second table shows all release exceptions written and whether they were submitted after or before the deployment.

The page looks like the current releases show page.

### CODE REVIEW

**Comment by popovm** [comment]
Ping @0mega and @bliof for feedback 👀

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -29,4 +29,11 @@ def path_from_url(url_or_path)
   rescue URI::InvalidURIError
     nil
   end
+
+  protected
+
+  def update_region_cookies
```
Comment (0mega) [inline app/controllers/application_controller.rb]: 🙇

File: app/controllers/unapproved_deployments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  def region
+    @region = cookies[:deploy_region]
+  end
+
+  def git_repository
+    git_repository_loader.load(app_name)
+  end
+
+  def force_html_format
```
Comment (0mega) [inline app/controllers/unapproved_deployments_controller.rb]: Is this used anywhere?

File: app/models/repositories/deploy_repository.rb [hunk 3]
```diff
@@ -14,6 +14,18 @@ def deploys_for(apps: nil, server:, at: nil)
       deploys(apps, server, at)
     end
 
+    def unapproved_production_deploys_for(app_name:, region:, from_date: nil, to_date: nil)
+      time_period_query = if from_date.present? && to_date.present?
+                            store.arel_table['deployed_at'].between(from_date.beginning_of_day..to_date.end_of_day)
+                          end
+      store
+        .where(time_period_query)
```
Comment (0mega) [inline app/models/repositories/deploy_repository.rb]: What will happen when `time_period_query` is nil, is it just ignored?

File: app/models/repositories/release_exception_repository.rb [hunk 4]
```diff
@@ -8,6 +8,17 @@ def initialize(store = Snapshots::ReleaseException)
       @store = store
     end
 
+    def release_exception_for_application(app_name:, from_date: nil, to_date: nil)
+      time_period_query = if from_date.present? && to_date.present?
+                            store.arel_table['submitted_at'].between(from_date.beginning_of_day..to_date.end_of_day)
+                          end
+
+      store
+        .where(time_period_query)
```
Comment (0mega) [inline app/models/repositories/release_exception_repository.rb]: ditto

File: app/controllers/unapproved_deployments_controller.rb (hunk 2 above)
Comment (popovm) [inline app/controllers/unapproved_deployments_controller.rb]: I'll add it to `application_controller.rb`

File: app/models/repositories/deploy_repository.rb (hunk 3 above)
Comment (popovm) [inline app/models/repositories/deploy_repository.rb]: It is ignored 👍 
Supported by a test: spec/models/repositories/deploy_repository_spec.rb:290

File: app/models/repositories/release_exception_repository.rb (hunk 4 above)
Comment (popovm) [inline app/models/repositories/release_exception_repository.rb]: It is ignored as above 👍 
Supported by a test: spec/models/repositories/release_exception_repository_spec.rb:18

File: app/controllers/unapproved_deployments_controller.rb (hunk 2 above)
Comment (popovm) [inline app/controllers/unapproved_deployments_controller.rb]: FYI: That's required, because of a really funny bug described and solved here: #127

**Comment by popovm** [comment]
Thank you for the review, @0mega 🙇 
I addressed your comments, could take a new look? 😼

File: app/views/unapproved_deployments/show.html.haml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    %tr
+      %th{width: '10%'} Version
+      %th{width: '10%'} Tickets
+      %th{width: '10%'} Feature Reviews
+      %th{width: '20%'} Deployed at
+      %th{width: '10%'} Deployed by
+      %th{width: '40%'} Deploy Alert
+  %tbody
+  - @unapproved_deploys.each do |deploy|
+    %tr.pending-release.danger
```
Comment (bliof) [inline app/views/unapproved_deployments/show.html.haml]: It will be better if we don't add the danger class if the review is approved.

**Pull request merged**